### DISCUSSION TITLE
REST Api for Valve

### DISCUSSION THREAD

**Discussion by DeepflashX** [Flo-Schilli/homeassistant-grohe_sense#42] [General]
Hey Flo,

I am trying to switch the Grohe Valve via REST Api.
But I have no clue how the url should look like.
Do you have an idea? The entity does not seem to be actually a switch, but I don*t know.

**Comment by Flo-Schilli** [comment]
Hey @DeepflashX 

I just shut down my PC but you have to send a POST Request to the following URL: 
'{self.__api_url}/locations/{location_id}/rooms/{room_id}/appliances/{appliance_id}/command'

I need to check tomorrow for the body data to switch it.

In Home Assistant it is defined as a valve entity, so matching to what it is 😊

**Comment by DeepflashX** [comment]
Is that the Grohe api or the HA Rest Api?
I need the HA Rest Api..

**Reply by Flo-Schilli** [reply]
Yes its the one from Grohe. About the HA I have no clue as I didn't needed it. Maybe you get some more info in the HA community for general valves and how this is done.

**Reply by DeepflashX** [reply]
thx for your feedback.
On the Readme you are referring to it also as a switch (and be careful about switching group.all_switches or so). 
Maybe that is a leftover because it was a switch-entity in the past.
To me it looks like it should be /api/services/valve/open or /api/services/valve/close but that does not work for me.
Could be also something else though. 
Is there a specific reason to make it a valve and not a switch? For switches there is more documentation regarding REST api..

**Reply by Flo-Schilli** [reply]
Yeah, at the beginning it was a switch, but when reworking most of the stuff I stumbled across the valve definition from HA and modified it due to this as it is specifically for this use case.
Therefore its just a not updated readme from my side - I will do this after the refactoring.

I guess I would try with this:
**POST /api/state/entity_id**
```json
{
    "state": "open"
}
```

**Comment by DeepflashX** [comment]
does not work unfortunately.. the target will be over the actions like /api/services.. 
Btw, could you add a simple 0 / 1 Status-Signal for the Valve? I am also having a hard time bringing the Valves Status into Loxone.
All analog values work without problems..

**Reply by DeepflashX** [reply]
ok nvm I created an analog sensor for that issue:
template:
  - sensor:
      - name: "Valve Status"
        unique_id: "valve_status_sensor"
        state: >
          {% if is_state('valve.mein_guard_valve', 'open') %}
            1
          {% elif is_state('valve.mein_guard_valve', 'close') %}
            0

**Comment by Vlada3333** [comment]
does not work

**Reply by Flo-Schilli** [reply]
What does not work? With little information like this no one can help you 🤷
